### ISSUE TITLE
function-name doesn't work with variabilized function name

### ISSUE THREAD

**Issue by dpetit62** [microsoft/tslint-microsoft-contrib#888]
### Bug Report

-   **`tslint-microsoft-contrib` version**: 5.2.1
-   **TSLint version**: 5.18.0
-   **TypeScript version**: 3.5.3
-   **Running TSLint via**: CLI (but occurs in VSCode too) 

#### TypeScript code being linted

```tsx
export const SET_USER = 'setUser';
export const mutations = {

  [SET_USER](state, user) {
        //
  }
};
```

with `tslint.json` configuration:

```json
'function-name': [
      true,
      {
        'function-regex': /^[a-z$][\w\d]+$/
      }
]

```

#### Actual behavior
const SET_USER: "setUser"
Method name does not match /^[a-z$][\w\d]+$/: [SET_USER] (function-name)tslint(function-name)

#### Expected behavior
The lint have to be ok because constant value correctly match the rule

**Comment by IllusionMH** [comment]
This rule doesn't use types and therefore can't verify value of a constant. Moreover, computed names can contain any expression/symbol etc. and can't be validated in general case. Therefore rule has code that should skip method declarations using computed names.
https://github.com/microsoft/tslint-microsoft-contrib/blob/bfc28e95dde1fab92aa457e79d338fb03b81d7cc/src/functionNameRule.ts#L217-L218

As I can see - you are using rather old version 5.2.1, while skipping computed names was added in 6.0.0 (See #459). Could you update your `tslint-microsoft-contrib` version and see if problem is still there?

**Comment by dpetit62** [comment]
The update to 6.0.0 solves the problem.

Thanks for your advice.

Just for information (for people who could have the same issue) :
In my project `tslint-microsoft-contrib` is a transitive depency, comming from `tslint-config-airbnb` which forces `tslint-microsoft-contrib` to be in ~5.2.1. You have to install `tslint-microsoft-contrib` as dev dependency in your project and to add `tslint-microsoft-contrib/latest` in the `extends` section of your `tslint.json` to override
